twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -281,99 +320,158 @@ def _scrooge_aspect_impl(target, ctx):
                 for d in ctx.rule.attr.deps
             ] + [target_ti],
         )
+    imps = [j[JavaInfo] for j in ctx.attr._implicit_compile_deps]
+
+    return (
+        target_ti,
+        transitive_ti,
+        deps,
+        imps,
+    )
+
+def _compile_thrift_to_language(target_ti, transitive_ti, language, target, ctx):
+    """Calls scrooge to compile thrift to the language specified in `language`.
+    Returns the name of the compiled jar."""
+
+    scrooge_file = ctx.actions.declare_file(
+        target.label.name + "_scrooge_{}.srcjar".format(language),
+    )
 
     # we sort so the inputs are always the same for caching
     compile_thrifts = sorted(target_ti.srcs.to_list())
-    imps = [j[JavaInfo] for j in ctx.attr._implicit_compile_deps]
-    if compile_thrifts:
-        # we sort so the inputs are always the same for caching
```

Comment:
I think this comment was lost

Reply:
I don't think so, it was duplicated to begin with :) it's here https://github.com/bazelbuild/rules_scala/pull/1055/files/c346443ab84f4f60b6178bdd2899102951d3519b#diff-4fe70c7ad76a7189b88d4792f15a77ffR340